### PR TITLE
Add handling of HardSigmoid recurrent activation for Keras LSTM

### PR DESCRIPTION
**Pull request by q-ycong-p**
This commit adds pattern matching and parsing where Keras LSTM uses
HardSigmoid as the recurrent activation. Refactored to allow addition
of other activations when needed.

Signed-off-by: Yu Cong <congyc@amazon.com>

### CODE REVIEW

**Comment by q-ycong-p** [comment]
The existing tf2 lstm rewriter exclusively handles {sigmoid,tanh,relu} as {recurrent,}activations. There are [other activations](https://www.tensorflow.org/api_docs/python/tf/keras/activations) defined in `tf.keras.activations`. For example, a `tf.keras.layers.LSTM` layer's recurrent activation may be `hard_sigmoid`. tf2onnx frozen graph alters patterns to represent the hard sigmoid computation as low-level ops (see attached screenshot). This messes existing pattern matching and fails tf2 lstm rewriter. This PR extends support to handle this variation, and refactors pattern matching to allow easy plug-in of other activation patterns in the future (i.e. ad hoc as needed).

See below for minimal example to reproduce the original issue. Prior to this PR, loop op is produced from Keras LSTM layer. This PR extends rewriter pattern matching to convert LSTM node correctly.
```
import tensorflow as tf
import tf2onnx

input = tf.keras.Input(shape=(10, 8))
lstm_output = tf.keras.layers.LSTM(4, recurrent_activation="hard_sigmoid")(input)
model = tf.keras.Model(input, lstm_output)
onnx_model, _ = tf2onnx.from_keras(model) # onnx model contains loop instead of LSTM op
```

Screen Shot 2022-07-15 at 2 23 29 PM

**Review by hwangdeyu — APPROVED** [review]
LGTM, thanks for your contribution! 

The `HardSigmoid` as the recurrent activation would be very helpful. Could you also tell us which model you are using to come up with this issue?

**Comment by q-ycong-p** [comment]
Hi @hwangdeyu , thank you for taking a look. The original model is internal. The general architecture is a simple LSTM encoder. If all is good, would you mind merging this change? Thanks!
